vignettes/extract-bias.Rmd

```diff
@@ -0,0 +1,494 @@
+(e.g. 5th and 95th, or 2.5th and 97.5th). However, in a few instances, only asymmetrical
+percentiles are available. We test whether asymmetry to varying degrees influences the
+bias of parameter extraction for all distributions.
+
+We set up the parameter space to explore:
+
+```{r, set-up-parameter-space-percentiles}
+distributions <- c("gamma", "lnorm", "weibull")
+dist_parameters <- seq(0.5, 2, 0.5)
+lower_percentiles <- c(2.5, 5, 25, 40)
+upper_percentiles <- c(60, 95, 97.5)
+
+parameters_perc <- expand.grid(
+  dist = distributions,
+  param_1 = dist_parameters,
+  param_2 = dist_parameters,
+  lower = lower_percentiles,
+  upper = upper_percentiles
+)
+
+# calculate the degree of asymmetry for each percentile combination
+lw_interval_diff <- abs(0 - parameters_perc$lower)
+up_interval_diff <- abs(100 - parameters_perc$upper)
+deg_asym <- abs(lw_interval_diff - up_interval_diff)
+
+# add degree of asymmetry to percentiles
+parameters_perc <- cbind(parameters_perc, deg_asym)
+
+# divide percentiles by 100 to make them probabilities for quantile functions
+parameters_perc$lower <- parameters_perc$lower / 100
+parameters_perc$upper <- parameters_perc$upper / 100
+```
+
+Now we can run the extraction for each point in parameter space. We set a seed
+to control for stochasticity when estimating the parameters, however changing or
+removing the seed should not drastically change the results or interpretation.
+
+```{r, set-seed}
+set.seed(1)
+```
+
+```{r, run-extraction-percentile}
+estim_params <- vector("list", nrow(parameters_perc))
+# Loop through parameter space estimating parameters
+for (params_idx in seq_len(nrow(parameters_perc))) {
+
+  dist <- as.character(parameters_perc[params_idx, "dist"])
+  percen <- unname(unlist(parameters_perc[params_idx, c("lower", "upper")]))
+
+  if (dist == "lnorm") {
+    true_values <- do.call(
+      paste0("q", dist),
+      list(
+        p = percen,
+        meanlog = parameters_perc[params_idx, "param_1"],
+        sdlog = parameters_perc[params_idx, "param_2"]
+      )
+    )
+  } else {
+    true_values <- do.call(
+      paste0("q", dist),
+      list(
+        p = percen,
+        shape = parameters_perc[params_idx, "param_1"],
+        scale = parameters_perc[params_idx, "param_2"]
+      )
+    )
+  }
+
+  # message about stochastic optimisation suppressed
+  estim_params[[params_idx]] <- suppressMessages(
+    extract_param(
+      type = "percentiles",
+      values = true_values,
+      distribution = dist,
+      percentiles = percen
+    )
+  )
+}
+
+# combine results
+results <- cbind(parameters_perc, do.call(rbind, estim_params))
+
+colnames(results) <- c(
+  "dist", "param_1", "param_2", "lower", "upper",
+  "deg_asym", "estim_param_1", "estim_param_2"
+)
+
+# calculate absolute difference between true parameter and estimated value
+results <- cbind(
+  results,
+  diff_param_1 = abs(results$param_1 - results$estim_param_1),
```

Comment:
Thanks for noting. I've made an issue from this comment to log it and we can consider if it might be something we want to include in the vignette after the package is released.